### PR TITLE
feat(issue-priority): Add issue.priority as a supported search term in UI

### LINKED ISSUE #65097: Add backend support for `issue.priority` search term

### PR DESCRIPTION
**Pull request by snigdhas**
image

Backend changes  in https://github.com/getsentry/sentry/pull/65205

Fixes https://github.com/getsentry/sentry/issues/65097

### CODE REVIEW

File: static/app/stores/tagStore.tsx [hunk 1]
```diff
@@ -90,6 +90,12 @@ const storeConfig: TagStoreDefinition = {
         ],
         predefined: true,
       },
+      [FieldKey.ISSUE_PRIORITY]: {
```
Comment (malwilley) [inline static/app/stores/tagStore.tsx:93]: We should only add this when the `issue-priority-ui` flag is enabled. Looks like we have the org here so should be a simple fix

File: static/app/utils/fields/index.ts [hunk 2]
```diff
@@ -1575,16 +1583,22 @@ export const getFieldDefinition = (
...
+    fieldKeys
+      .map(fieldKey => [
+        fieldKey,
+        {
+          key: fieldKey,
+          name: fieldKey,
+          kind: getFieldDefinition(fieldKey)?.kind,
+        },
+      ])
+      .filter(([key, _]) => (key === FieldKey.ISSUE_PRIORITY ? include_priority : true))
```
Comment (malwilley) [inline static/app/utils/fields/index.ts:1601]: I don't think we actually need to filter it out here, looking at the call site this function won't ever get passed `ISSUE_PRIORITY`

File: static/app/views/alerts/wizard/options.tsx [hunk 3]
```diff
@@ -267,7 +267,7 @@ export function datasetSupportedTags(
       [Dataset.SESSIONS]: SESSION_SUPPORTED_TAGS,
     },
     value => {
-      return value ? makeTagCollection(value) : undefined;
+      return value ? makeTagCollection(value, org) : undefined;
```
Comment (malwilley) [inline static/app/views/alerts/wizard/options.tsx:270]: Same comment as above

File: static/app/views/issueList/searchBar.tsx [hunk 4]
```diff
@@ -23,18 +24,22 @@ import usePageFilters from 'sentry/utils/usePageFilters';
...
+      .map(key => [
+        key,
+        {
+          ...supportedTags[key],
+          kind:
+            getFieldDefinition(key)?.kind ??
+            (supportedTags[key].predefined ? FieldKind.FIELD : FieldKind.TAG),
+        },
+      ])
+      .filter(([key, _]) => (key === FieldKey.ISSUE_PRIORITY ? include_priority : true))
```
Comment (malwilley) [inline static/app/views/issueList/searchBar.tsx:40]: I also don't think it's necessary to filter here - the modification you added in `tagStore.tsx` should have already filtered it out from `tags`

Comment (snigdhas) [reply static/app/views/issueList/searchBar.tsx:40]: Removing this ends up showing `issue.priority` in the search

Comment (malwilley) [reply static/app/views/issueList/searchBar.tsx:40]: Hmm I must be missing something. Looks good then!

**Review by malwilley — APPROVED** [review]

**Pull request merged**